server/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -343,6 +377,9 @@ public void onStoreClosed(ShardId shardId) {
         );
         boolean success = false;
         try {
+            if (writeDanglingIndices && nodeWriteDanglingIndicesInfo) {
+                indexService.addMetadataListener(imd -> updateDanglingIndicesInfo(index));
+            }
             indexService.getIndexEventListener().afterIndexCreated(indexService);
             indices = newMapBuilder(indices).put(index.getUUID(), indexService).immutableMap();
```

Comment:
In ES there's also a 

```java
            if (writeDanglingIndices) {
                if (nodeWriteDanglingIndicesInfo) {
                    updateDanglingIndicesInfo(index);
                } else {
                    indexService.deleteDanglingIndicesInfo();
                }
            }
```

here. Seems like this might be missing?